<launch>
    <!-- ODOM = 0, POSE = 1, GAZEBO = 2,  MOCAP = 3, VIOBOT = 4, GPS = 5, RTK = 6, VINS = 7 -->
    <arg name="uav_id" default="1" />
    <arg name="external_source" default="0" />
    <arg name="uav_name" default="uav" />
    <arg name="position_topic" default="/sunray/odometry" />
    <arg name="enable_rviz" default="false" />
    <arg name="enable_range_sensor" default="false" />
    <!-- 是否使用vision_pose话题至PX4，false:直接使用Mavlink发送外部定位数据到PX4 -->
    <arg name="use_vision_pose" default="true" />
    <arg name="server" default="192.168.20.15"/>
    <arg name="tilted" default="false"/>
    <!-- 添加launch-prefix 用于一键脚本中新开终端 -->
    <arg name="launch-prefix" default="" />
    <!-- RTK模式：原点设置方式 -->
    <arg name="rtk_origin_mode" default="1" />  <!-- 原点设置模式: 0=launch文件, 1=地面站设置 -->
    
    <!-- RTK模式：ENU坐标转换原点参数（仅在RTK模式下使用） -->
    <arg name="rtk_origin_lat" default="47.3977420" />  <!-- 原点纬度，单位：[°] -->
    <arg name="rtk_origin_lon" default="8.54559400" />  <!-- 原点经度，单位：[°] -->
    <arg name="rtk_origin_alt" default="488.01" />      <!-- 原点高度，单位：[m] -->

    <!-- 启动 external_fusion_node-->
    <node name="external_fusion" pkg="sunray_uav_control" type="external_fusion_node" output="screen" launch-prefix="$(arg launch-prefix)">
        <param name="uav_id" value="$(arg uav_id)" />
        <param name="external_source" value="$(arg external_source)" />
        <param name="uav_name" value="$(arg uav_name)" />
        <param name="position_topic" value="$(arg position_topic)" />
        <param name="enable_range_sensor" value="$(arg enable_range_sensor)" />
        <param name="use_vision_pose" value="$(arg use_vision_pose)" />
        <param name="tilted" value="$(arg tilted)" />
        <!-- RTK模式：原点设置方式 -->
        <param name="rtk_origin_mode" value="$(arg rtk_origin_mode)" />
        <!-- RTK模式：ENU坐标转换原点参数 -->
        <param name="rtk_origin_lat" value="$(arg rtk_origin_lat)" />
        <param name="rtk_origin_lon" value="$(arg rtk_origin_lon)" />
        <param name="rtk_origin_alt" value="$(arg rtk_origin_alt)" />
    </node>

    <!-- 启动 vrpn_client_ros -->
    <include file="$(find sunray_uav_control)/launch/sunray_vrpn.launch" if="$(eval external_source == 3)">
        <arg name="server" value="$(arg server)"/>
        <arg name="uav_id" value="$(arg uav_id)"/>
    </include>

    <arg name="rivz_config" default="$(find sunray_uav_control)/rviz/uav.rviz"/>
	<!-- 启动Rviz-->
	<group if="$(arg enable_rviz)">
        <node type="rviz" name="rviz_external_fusion" pkg="rviz" args="-d $(arg rivz_config)"/>
    </group>
</launch>
